Patterns/Italian/Italian-DateTime.yaml

```diff
@@ -288,283 +343,273 @@ TimeRegex6: !nestedRegex
+  references: [WeekDayRegex, FlexibleDayRegex, MonthRegex]
 WeekDayAndDayRegex: !nestedRegex
-  def: \b{WeekDayRegex}\s+(?!(the)){DayRegex}(?!([-]|(\s+({AmDescRegex}|{PmDescRegex|{OclockRegex}}))))\b
+  def: \b{WeekDayRegex}\s*,?\s+(?!(il|l')){DayRegex}(?!([-]|(\s+({AmDescRegex}|{PmDescRegex|{OclockRegex}}))))\b
   references: [WeekDayRegex, DayRegex, AmDescRegex, PmDescRegex, OclockRegex]
 RestOfDateRegex: !simpleRegex
-  def: Fine\s+(di\s+)?((del|questo|questa|corrente)\s+)?(?<duration>settimana|mese|anno)\b
+  def: \bresto\s+((di|del(l[a'])?)\s*)?((quest[oa']|corrente)\s*)?(?<duration>settimana|mese|anno|decennio)\b
 RestOfDateTimeRegex: !simpleRegex
-  def: Fine\s+(di\s+)?((del|questo|questa|corrente)\s+)?(?<unit>giorno)\b
-LaterEarlyPeriodRegex: !simpleRegex
-  # TODO: modify below regex according to the counterpart in English
-  def: ^[.]
-WeekWithWeekDayRangeRegex: !simpleRegex
-  # TODO: modify below regex according to the counterpart in English
-  def: ^[.]
+  def: \bresto\s+((di|del(la)?)\s+)?((quest[oa]|corrente)\s+)?(?<unit>giorn(o|ata))\b
+SuffixAfterRegex: !simpleRegex
+  def: \b(((a)\s)?(o|e)\s+(sopra|dopo|più\s+tardi|più\s+grande)(?!\s+di))\b
+LaterEarlyPeriodRegex: !nestedRegex
+  def: \b(((inizio|metà|fine)(\s+(di|del(l[ao'])))?)\s*\b\s*(?<suffix>{OneWordPeriodRegex})|({UnspecificEndOfRangeRegex}))\b
+  references: [OneWordPeriodRegex, UnspecificEndOfRangeRegex]
+WeekWithWeekDayRangeRegex: !nestedRegex
+  def: \b((?<week>(inizio(\s+della)?|fine(\s+della)?|questa)\s+settimana)((\s+[tf]ra\s+{WeekDayRegex}\s+e\s+{WeekDayRegex})|(\s+da\s+{WeekDayRegex}\s+a\s+{WeekDayRegex})))\b
+  references: [WeekDayRegex]
 GeneralEndingRegex: !simpleRegex
   def: ^\s*((\.,)|\.|,|!|\?)?\s*$
 MiddlePauseRegex: !simpleRegex
   def: \s*(,)\s*
 DurationConnectorRegex: !simpleRegex
   def: ^\s*(?<connector>\s+|e|,)\s*$
 PrefixArticleRegex: !simpleRegex
-  def: \bil\s+
+  def: \b(il?|l[oae']|gli)\s*
 OrRegex: !simpleRegex
-  def: \s*(o)\s*
+  def: \s*(o|oppure)\s*
 YearPlusNumberRegex: !nestedRegex
-  def: \b(Anno\s+((?<year>(\d{3,4}))|{FullTextYearRegex}))\b
+  def: \b(anno\s+((?<year>(\d{3,4}))|{FullTextYearRegex}))\b
   references: [ FullTextYearRegex ]
 NumberAsTimeRegex: !nestedRegex
   def: \b({EngTimeRegex}|{PeriodHourNumRegex}|{BaseDateTime.HourRegex})\b
   references: [ EngTimeRegex, PeriodHourNumRegex, BaseDateTime.HourRegex ]
 TimeBeforeAfterRegex: !nestedRegex
-  def: \b(((?<=\b(prima|non prima di|per|dopo)\s+)({EngTimeRegex}|{HourNumRegex}|{BaseDateTime.HourRegex}|{MidTimeRegex}))|{MidTimeRegex})\b
+  def: \b(((?<=\b(prima (di|dell[e'])|non prima (di|dell[e'])|per(\s+l[e'])?|dopo(\s+l[e'])?)\s*)({EngTimeRegex}|{HourNumRegex}|{BaseDateTime.HourRegex}|{MidTimeRegex}))|{MidTimeRegex})\b
   references: [ EngTimeRegex, HourNumRegex, BaseDateTime.HourRegex, MidTimeRegex ]
 DateNumberConnectorRegex: !simpleRegex
   def: ^\s*(?<connector>\s+il)\s*$
-CenturyRegex: !simpleRegex
```

Comment:
I'm not sure if I understand the reason for upraising these declarations.

Reply:
some regex have been moved up in the yaml file (e.g. CenturyRegex) because they were referenced before declaration